source/common/config/grpc_stream.h

```diff
@@ -85,6 +95,7 @@ class GrpcStream : public Grpc::AsyncStreamCallbacks<ResponseProto>,
   void onReceiveMessage(ResponseProtoPtr<ResponseProto>&& message) override {
     // Reset here so that it starts with fresh backoff interval on next disconnect.
     backoff_strategy_->reset();
+    unsetFailure();
```

Comment:
Why is this needed? Isn't it sufficient to unset on successful connect?

Reply:
It is the comment on the next line that lead me to believe I need a reset here, to.
But I think it makes sense to remove it. ack.